src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -630,6 +659,10 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
       return null;
     } else if (typeof device === 'string') {
       trackConstraints.deviceId = { exact: device };
+      if (kind === 'audio' && setGroupId === true) {
+        groupId = this.deviceIdGroupIdMap['audioinput'].get(device);
+        trackConstraints.groupId = groupId;
```

Comment:
This PR assumes that the `Device` device is a string ID; however, `Device` can be `string | MediaTrackConstraints | MediaStream | null`.

For example, this flow won't work.
1. `chooseAudioInputDevice('default')` to choose the default.
2. Connect a new device "Bluetooth Headphone." A new default is "Bluetooth Headphone."
3. Choose the new default by passing the constraints.
    ```
    chooseAudioInputDevice({
      deviceId: { exact: 'default' };
    })
    ```